grails-app/services/au/org/ala/specieslist/QueryService.groovy

```diff
@@ -442,14 +481,21 @@ class QueryService {
      * @param base
      * @param facets
      * @param dataResourceUid
+     * @param q
      * @return
      */
-    def constructWithFacets(String base, List facets, String dataResourceUid) {
+    def constructWithFacets(String base, List facets, String dataResourceUid, String q) {
```

Comment:
Could've use the groovy optional args/params here:
`def constructWithFacets(String base, List facets, String dataResourceUid, String q = "") {`

Reply:
or ` String q = null) {`

Reply:
That way, all the existing method calls in controller wouldn't have needed adding `, null)` at end.

Reply:
good point, can do that.